### PR TITLE
chore: update skipped flaky tests

### PR DESCRIPTION
**Pull request by Captainia**
*Issue #, if available:*

*Description of changes:*

1. Some of the JumpStart notebook_utils tests are skipped due to flakiness, and one of the test is failing consistently. This change fixes these tests.
2. Together, update the `_construct_payload` internal function to accept model_type, this is used by Margaret team internally.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/jumpstart/test_notebook_utils.py [hunk 1]
```diff
@@ -547,12 +541,15 @@ def vulnerable_training_model_spec(bucket, key, *args, **kwargs):
         patched_read_s3_file.side_effect = vulnerable_inference_model_spec
 
         num_specs = len(PROTOTYPICAL_MODEL_SPECS_DICT)
-        num_prop_specs = len(BASE_PROPRIETARY_MANIFEST)
         assert [] == list_jumpstart_models(
-            And("inference_vulnerable is false", "training_vulnerable is false")
+            And(
+                "inference_vulnerable is false",
+                "training_vulnerable is false",
+                "model_type is open_weights",
```
Comment (evakravi) [inline tests/unit/sagemaker/jumpstart/test_notebook_utils.py:548]: do we need to include "model_type is open_weights" or is it still included by default?

File: tests/unit/sagemaker/jumpstart/test_notebook_utils.py [hunk 2]
```diff
@@ -574,10 +575,6 @@ def vulnerable_training_model_spec(bucket, key, *args, **kwargs):
 
         assert patched_read_s3_file.call_count == 0
 
-    @pytest.mark.skipif(
```
Comment (evakravi) [inline tests/unit/sagemaker/jumpstart/test_notebook_utils.py:577]: i'm still confused why this was flakey in the first place and how your PR fixed this

**Review by evakravi — APPROVED** [review]

File: tests/unit/sagemaker/jumpstart/test_notebook_utils.py (hunk 1 above)
Comment (Captainia) [reply tests/unit/sagemaker/jumpstart/test_notebook_utils.py:548]: It is not included by default by design, `list_jumpstart_models` has both open weights and proprietary models for discoverability, and `"model_type is open_weights"` filter is used to only list open weights models.

File: tests/unit/sagemaker/jumpstart/test_notebook_utils.py (hunk 2 above)
Comment (Captainia) [reply tests/unit/sagemaker/jumpstart/test_notebook_utils.py:577]: I think this was flaky before we introduced prop models, but somehow seen failing more often after prop models. I updated to only list open weight models, and tried starting the test in multiple threads in parallel locally and they haven't been failing once for me. I think there is less parallelism in PR hooks now, so hope this is not flaky anymore.

**Review by zhaoqizqwang — APPROVED** [review]

**Pull request merged**